# Task 01 - Track A: Python Basics & Dataset Exploration

**Course:** Database Applications Development  
**Lesson:** 01 - Introduction to JupyterLab & Python Fundamentals  


---

## Instructions

Complete all exercises in this notebook. Each section includes example code to help guide you. Read the examples carefully, then complete the exercises below them.

**Submission:**
1. Save this notebook as `dbAppsTask01TrackA.ipynb`
2. Add, commit, and push to your `databaseApplications` repository on GitHub
3. Verify the file appears correctly on GitHub

---

## Titanic Dataset - Data Dictionary

You'll be working with the Titanic dataset throughout this task. Here's what each column means:

| Column | Description | Data Type | Example Values |
|--------|-------------|-----------|----------------|
| **pclass** | Passenger class (ticket class) | Integer | 1 = First Class<br>2 = Second Class<br>3 = Third Class |
| **survived** | Survival status | Integer | 0 = Did not survive<br>1 = Survived |
| **name** | Passenger's full name | String | "Braund, Mr. Owen Harris" |
| **sex** | Gender | String | "male" or "female" |
| **age** | Age in years | Float | 22.0, 38.0, 26.0 |
| **sibsp** | Number of siblings/spouses aboard | Integer | 0, 1, 2, etc. |
| **parch** | Number of parents/children aboard | Integer | 0, 1, 2, etc. |
| **ticket** | Ticket number | String | "A/5 21171", "PC 17599" |
| **fare** | Passenger fare (ticket price) | Float | 7.25, 71.28, 8.05 |
| **cabin** | Cabin number | String | "C85", "E46", "B96 B98" |
| **embarked** | Port of embarkation | String | C = Cherbourg<br>Q = Queenstown<br>S = Southampton |
| **boat** | Lifeboat number | String | "13", "4", "D" |
| **body** | Body identification number | Integer | For victims recovered |
| **home.dest** | Home/destination | String | "New York, NY", "Montreal, PQ" |

**Note:** Some columns may have missing values (NaN = Not a Number), which is common in real-world datasets.

---

## Part 1: Variables and Data Types

Variables store data for later use. Python has several built-in data types.

### Example: Creating Variables

In [4]:
# Example of different data types
student_name = "Alice Johnson"  # String (text)
student_age = 17                # Integer (whole number)
student_gpa = 3.75              # Float (decimal number)
is_passing = False              # Boolean (True/False)

# Print with descriptive labels
print("Name:", student_name)
print("Age:", student_age)
print("GPA:", student_gpa)
print("Passing:", is_passing)

Name: Alice Johnson
Age: 17
GPA: 3.75
Passing: False


### Exercise 1.1: Create Your Own Variables

Create the following variables with your own information:
- `my_name` - your full name (string)
- `my_age` - your age (integer)
- `my_gpa` - a GPA value (float)
- `enrolled` - whether you're enrolled in this course (boolean)

Print each variable with a descriptive label.

In [11]:
# Your code here
my_name = "Gio"
my_age = 17
my_gpa = 3.86
enrolled = True

# Print statements
print("name: ", my_name)
print("age: ", my_age)
print("gpa: ", my_gpa)
print("passing: ", enrolled)

name:  Gio
age:  17
gpa:  3.86
passing:  True


### Example: Checking Data Types

In [13]:
# Use type() to check what kind of data is stored
course_name = "Database Applications"
course_code = 145085.1

print("Type of course_name:", type(course_name))  # <class 'str'>
print("Type of course_code:", type(course_code))  # <class 'int'>

Type of course_name: <class 'str'>
Type of course_code: <class 'float'>


### Exercise 1.2: Check Data Types

Use the `type()` function to check the data type of each variable you created in Exercise 1.1. Print the results.

In [16]:
print("Type of name:", type(my_name))  
print("Type of age:", type(my_age))  
print("Type of gpa:", type(my_gpa))  
print("Type of enrolled:", type(enrolled))  



Type of name: <class 'str'>
Type of age: <class 'int'>
Type of gpa: <class 'float'>
Type of enrolled: <class 'bool'>


---

## Part 2: Basic Operations

Python can perform calculations and manipulate strings.

### Example: Mathematical Operations

In [22]:
# Calculate total cost for movie tickets
ticket_price = 10.50
number_of_tickets = 4

total_cost = ticket_price * number_of_tickets

print("Total cost for", number_of_tickets, "tickets: $", total_cost)

Total cost for 4 tickets: $ 42.0


### Exercise 2.1: Calculate Ticket Prices

A theater charges:
- Adult tickets: $12.50
- Child tickets: $7.50

Calculate the total cost for a family with:
- 2 adults
- 3 children

Store the result in a variable called `total_cost` and print it with a descriptive message.

In [37]:
# Your code here
adult_price = 12.50
child_price = 7.50
num_adults = 2
num_children = 3
number_of_tickets = num_adults + num_children

total_cost = adult_price * num_adults + child_price * num_children


print("Total cost for", number_of_tickets, "tickets: $", total_cost)

Total cost for 5 tickets: $ 47.5


### Example: String Concatenation

In [26]:
# Combining strings together
first = "John"
last = "Smith"
age = 25

# Method 1: Using + operator
full_name = first + " " + last
print(full_name)

# Method 2: Using f-strings (formatted string literals)
message = f"Hello! My name is {first} {last} and I am {age} years old."
print(message)

John Smith
Hello! My name is John Smith and I am 25 years old.


### Exercise 2.2: String Concatenation

Create three variables:
- `first_name`
- `last_name`
- `favorite_subject`

Combine them to create and print a sentence like:  
"Hello! My name is [first] [last] and my favorite subject is [subject]."

You can use either the + operator or f-strings.

In [1]:
# Your code here
first_name = "Gio"
last_name = "Adame"
favorite_subject = "Math"

sentence =  "Hello! My name is [first] [last] and my favorite subject is [subject]."
print(sentence)


Hello! My name is [first] [last] and my favorite subject is [subject].


### Exercise 2.2: String Concatenation

Create three variables:
- `first_name`
- `last_name`
- `favorite_subject`

Combine them to create and print a sentence like:  
"Hello! My name is [first] [last] and my favorite subject is [subject]."

You can use either the + operator or f-strings.

---

## Part 3: Working with Comments

Comments explain what your code does. They're ignored by Python but help humans understand your work.

### Example: Using Comments

In [2]:
# Calculate the area of a circle
# Formula: area = π × radius²

radius = 5          # Circle radius in centimeters
pi = 3.14159        # Approximation of π (pi)

area = pi * radius ** 2  # ** means "to the power of"

print("Area of circle:", area, "square cm")

Area of circle: 78.53975 square cm


### Exercise 3.1: Add Comments

The code below calculates the area of a rectangle. Add comments to explain each step.

In [4]:
length = 10
width = 5
# Calculate the area by multiplying the 'length' and 'width', 
# and assign the result to the variable 'area'
area = length * width
# Print the final result (the area) to the console
print("The area is:", area)


The area is: 50


---

## Part 4: Introduction to Pandas

Pandas is Python's most powerful library for working with data. We'll use it to load and explore the Titanic dataset.

### Example: Importing Pandas

In [5]:
# Import pandas with the standard alias 'pd'
import pandas as pd

# Now we can use pandas functions by typing pd.function_name()
print("Pandas version:", pd.__version__)

Pandas version: 2.2.3


### Exercise 4.1: Import Pandas

Import the pandas library using the standard alias `pd`.

In [2]:
import pandas as pd
print("Pandas version:", pd.__version__)

Pandas version: 2.2.3


### Example: Loading a CSV File

In [ ]:
# Load a CSV file into a DataFrame
# Replace 'sample.csv' with your actual filename
data = pd.read_csv('sample.csv')

# The data is now stored in a DataFrame object
# A DataFrame is like a spreadsheet - it has rows and columns

### Exercise 4.2: Load the Titanic Dataset

Use pandas to read the `Titanic Dataset.csv` file into a DataFrame called `titanic`.

**Note:** Make sure the CSV file is in the same folder as this notebook!

In [5]:
import pandas as pd
titanic = pd.read_csv('Titanic Dataset.csv')
print("First 5 rows of the 'titanic' DataFrame:")
print(titanic.head())


First 5 rows of the 'titanic' DataFrame:
   pclass  survived                                             name     sex  \
0       1         1                    Allen, Miss. Elisabeth Walton  female   
1       1         1                   Allison, Master. Hudson Trevor    male   
2       1         0                     Allison, Miss. Helen Loraine  female   
3       1         0             Allison, Mr. Hudson Joshua Creighton    male   
4       1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

     age  sibsp  parch  ticket      fare    cabin embarked boat   body  \
0  29.00      0      0   24160  211.3375       B5        S    2    NaN   
1   0.92      1      2  113781  151.5500  C22 C26        S   11    NaN   
2   2.00      1      2  113781  151.5500  C22 C26        S  NaN    NaN   
3  30.00      1      2  113781  151.5500  C22 C26        S  NaN  135.0   
4  25.00      1      2  113781  151.5500  C22 C26        S  NaN    NaN   

                         home.des

### Exercise 4.2: Load the Titanic Dataset

Use pandas to read the `Titanic Dataset.csv` file into a DataFrame called `titanic`.

**Note:** Make sure the CSV file is in the same folder as this notebook!

### Example: Viewing the First Rows

In [ ]:
# View the first 5 rows of any DataFrame
data.head()

# You can specify how many rows to show
data.head(10)  # Shows first 10 rows

### Exercise 4.3: View the First Rows

Display the first 5 rows of the Titanic dataset using the `.head()` method.

In [7]:
titanic.head(10)

,pclass,survived,name,sex,age,sibsp,parch,ticket,fare,cabin,embarked,boat,body,home.dest
0,1,1,"Allen, Miss. Elisabeth Walton",female,29.00,0,0,24160,211.3375,B5,S,2,NaN,"St Louis, MO"
1,1,1,"Allison, Master. Hudson Trevor",male,0.92,1,2,113781,151.5500,C22 C26,S,11,NaN,"Montreal, PQ / Chesterville, ON"
2,1,0,"Allison, Miss. Helen Loraine",female,2.00,1,2,113781,151.5500,C22 C26,S,NaN,NaN,"Montreal, PQ / Chesterville, ON"
3,1,0,"Allison, Mr. Hudson Joshua Creighton",male,30.00,1,2,113781,151.5500,C22 C26,S,NaN,135.0,"Montreal, PQ / Chesterville, ON"
4,1,0,"Allison, Mrs. Hudson J C (Bessie Waldo Daniels)",female,25.00,1,2,113781,151.5500,C22 C26,S,NaN,NaN,"Montreal, PQ / Chesterville, ON"
5,1,1,"Anderson, Mr. Harry",male,48.00,0,0,19952,26.5500,E12,S,3,NaN,"New York, NY"
6,1,1,"Andrews, Miss. Kornelia Theodosia",female,63.00,1,0,13502,77.9583,D7,S,10,NaN,"Hudson, NY"
7,1,0,"Andrews, Mr. Thomas Jr",male,39.00,0,0,112050,0.0000,A36,S,NaN,NaN,"Belfast, NI"
8,1,1,"Appleton, Mrs. Edward Dale (Charlotte Lamson)",female,53.00,2,0,11769,51.4792,C101,S,D,NaN,"Bayside, Queens, NY"
9,1,0,"Artagaveytia, Mr. Ramon",male,71.00,0,0,PC 17609,49.5042,NaN,C,NaN,22.0,"Montevideo, Uruguay"


### Example: Dataset Information

In [ ]:
# Get information about the DataFrame structure
data.info()

# This shows:
# - Number of rows and columns
# - Column names
# - Data types of each column
# - How many non-null (non-missing) values in each column

### Exercise 4.4: Dataset Information

Use the `.info()` method to see the structure of the Titanic dataset.

In [8]:
titanic.info(10)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 14 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   pclass     1309 non-null   int64  
 1   survived   1309 non-null   int64  
 2   name       1309 non-null   object 
 3   sex        1309 non-null   object 
 4   age        1046 non-null   float64
 5   sibsp      1309 non-null   int64  
 6   parch      1309 non-null   int64  
 7   ticket     1309 non-null   object 
 8   fare       1308 non-null   float64
 9   cabin      295 non-null    object 
 10  embarked   1307 non-null   object 
 11  boat       486 non-null    object 
 12  body       121 non-null    float64
 13  home.dest  745 non-null    object 
dtypes: float64(3), int64(4), object(7)
memory usage: 143.3+ KB


### Exercise 4.5: Answer Questions

Based on the `.info()` output, answer these questions in the markdown cell below:

1. How many rows (passengers) are in the dataset?
2. How many columns are in the dataset?
3. What is the data type of the 'age' column?
4. What is the data type of the 'survived' column?

**Your Answers:**

1. Number of rows: 
2. Number of columns: 
3. Data type of 'age': 
4. Data type of 'survived': 

---

## Part 5: Basic DataFrame Exploration

Let's explore the Titanic dataset using pandas methods.

### Example: Common DataFrame Methods

In [ ]:
# Useful DataFrame methods:

data.tail()       # View last rows
data.columns      # Get column names
data.shape        # Get (rows, columns) as a tuple
data.describe()   # Get statistics for numerical columns

### Exercise 5.1: View Last Rows

Display the last 5 rows of the dataset using the `.tail()` method.

In [10]:
titanic.tail(10)


,pclass,survived,name,sex,age,sibsp,parch,ticket,fare,cabin,embarked,boat,body,home.dest
1299,3,0,"Yasbeck, Mr. Antoni",male,27.0,1,0,2659,14.4542,NaN,C,C,NaN,NaN
1300,3,1,"Yasbeck, Mrs. Antoni (Selini Alexander)",female,15.0,1,0,2659,14.4542,NaN,C,NaN,NaN,NaN
1301,3,0,"Youseff, Mr. Gerious",male,45.5,0,0,2628,7.2250,NaN,C,NaN,312.0,NaN
1302,3,0,"Yousif, Mr. Wazli",male,NaN,0,0,2647,7.2250,NaN,C,NaN,NaN,NaN
1303,3,0,"Yousseff, Mr. Gerious",male,NaN,0,0,2627,14.4583,NaN,C,NaN,NaN,NaN
1304,3,0,"Zabour, Miss. Hileni",female,14.5,1,0,2665,14.4542,NaN,C,NaN,328.0,NaN
1305,3,0,"Zabour, Miss. Thamine",female,NaN,1,0,2665,14.4542,NaN,C,NaN,NaN,NaN
1306,3,0,"Zakarian, Mr. Mapriededer",male,26.5,0,0,2656,7.2250,NaN,C,NaN,304.0,NaN
1307,3,0,"Zakarian, Mr. Ortin",male,27.0,0,0,2670,7.2250,NaN,C,NaN,NaN,NaN
1308,3,0,"Zimmerman, Mr. Leo",male,29.0,0,0,315082,7.8750,NaN,S,NaN,NaN,NaN


### Exercise 5.2: Get Column Names

Print all the column names in the dataset using `.columns`.

In [11]:
# Your code here
titanic.columns

Index(['pclass', 'survived', 'name', 'sex', 'age', 'sibsp', 'parch', 'ticket',
       'fare', 'cabin', 'embarked', 'boat', 'body', 'home.dest'],
      dtype='object')

### Exercise 5.3: Get Dataset Shape

Print the shape (rows, columns) of the dataset using `.shape`.

**Hint:** This will return a tuple like (1309, 14) meaning 1309 rows and 14 columns.

In [12]:
# Your code here
titanic.shape

(1309, 14)

### Example: Descriptive Statistics

In [ ]:
# Get basic statistics about numerical columns
data.describe()

# This shows:
# count - number of non-missing values
# mean - average value
# std - standard deviation
# min - minimum value
# 25% - first quartile
# 50% - median (middle value)
# 75% - third quartile
# max - maximum value

### Exercise 5.4: Basic Statistics

Use the `.describe()` method to see basic statistics about numerical columns in the Titanic dataset.

In [15]:
# Your code here
titanic.describe(1       c          )

TypeError: 'int' object is not iterable

### Exercise 4.2: Load the Titanic Dataset

Use pandas to read the `Titanic Dataset.csv` file into a DataFrame called `titanic`.

**Note:** Make sure the CSV file is in the same folder as this notebook!

### Exercise 5.5: Interpret Statistics

Based on the `.describe()` output, answer these questions:

1. What is the average (mean) age of passengers?
2. What is the maximum fare paid?
3. What percentage of passengers survived? (Hint: Look at the 'survived' column mean - it will be a decimal between 0 and 1)

**Your Answers:**

1. Average age: 
2. Maximum fare: 
3. Survival rate (as percentage): 

---

## Submission Checklist

Before submitting, make sure you have:

- [ ] Completed all exercises
- [ ] Run all cells successfully (no errors)
- [ ] Added your name and date at the top
- [ ] Saved the notebook as `dbAppsTask01TrackA.ipynb`
- [ ] Pushed to your `databaseApplications` repository on GitHub
- [ ] Verified the file appears on GitHub

**Great work on your first database applications task!**